dev-libs/efl/efl-1.20.7.ebuild

```diff
@@ -0,0 +1,326 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="5"
+
+MY_P=${P/_/-}
+
+if [[ "${PV}" == "9999" ]] ; then
+	EGIT_SUB_PROJECT="core"
+	EGIT_URI_APPEND="${PN}"
+elif [[ *"${PV}" == *"_pre"* ]] ; then
+	MY_P=${P%%_*}
+	SRC_URI="https://download.enlightenment.org/pre-releases/${MY_P}.tar.xz"
+else
+	SRC_URI="https://download.enlightenment.org/rel/libs/${PN}/${MY_P}.tar.xz"
+	KEYWORDS="~amd64 ~arm ~hppa ~mips ~ppc ~ppc64 ~sh ~sparc ~x86 ~amd64-linux ~x86-linux ~ppc-macos ~x86-macos ~x64-solaris ~x86-solaris"
+fi
+
+inherit enlightenment pax-utils gnome2-utils xdg-utils
+
+DESCRIPTION="Enlightenment Foundation Libraries all-in-one package"
+
+LICENSE="BSD-2 GPL-2 LGPL-2.1 ZLIB"
+IUSE="avahi +bmp debug connman drm +eet egl examples fbcon +fontconfig fribidi gif gles glib gnutls gstreamer harfbuzz hyphen +ico ibus jpeg2k libressl neon oldlua opengl ssl pdf physics pixman +png postscript +ppm +psd pulseaudio raw scim sdl sound svg systemd tga tiff tslib unwind v4l valgrind vlc vnc wayland webp X xcf xim xine xpm"
+
+REQUIRED_USE="
+	pulseaudio?	( sound )
+	opengl?		( || ( X sdl wayland ) )
+	gles?		( || ( X wayland ) )
+	gles?		( !sdl )
+	gles?		( egl )
+	ibus?		( glib )
+	sdl?		( opengl )
+	vnc?		( X fbcon )
+	wayland?	( egl !opengl gles )
+	xim?		( X )
+"
+
+RDEPEND="
+	avahi? ( net-dns/avahi )
+	connman? ( net-misc/connman )
+	drm? (
+		>=dev-libs/libinput-0.8
+		media-libs/mesa[gbm]
+		>=x11-libs/libxkbcommon-0.3.0
+	)
+	fontconfig? ( >=media-libs/fontconfig-2.5.0 )
+	fribidi? ( >=dev-libs/fribidi-0.19.2 )
+	gif? ( media-libs/giflib )
+	glib? ( dev-libs/glib:2 )
+	gnutls? ( >=net-libs/gnutls-3.3.6 )
+	!gnutls? (
+		ssl? (
+			!libressl? ( dev-libs/openssl:0= )
+			libressl? ( dev-libs/libressl )
+		)
+	)
+	gstreamer? (
+		media-libs/gstreamer:1.0
+		media-libs/gst-plugins-base:1.0
+	)
+	harfbuzz? ( >=media-libs/harfbuzz-0.9.0 )
+	hyphen? ( dev-libs/hyphen )
+	ibus? ( >=app-i18n/ibus-1.4 )
+	jpeg2k? ( media-libs/openjpeg:0 )
+	!oldlua? ( >=dev-lang/luajit-2.0.0 )
+	oldlua? ( dev-lang/lua:* )
+	pdf? ( app-text/poppler[cxx] )
+	physics? ( >=sci-physics/bullet-2.80 )
+	pixman? ( x11-libs/pixman )
+	postscript? ( app-text/libspectre )
+	png? ( media-libs/libpng:0= )
+	pulseaudio? ( media-sound/pulseaudio )
+	raw? ( media-libs/libraw )
+	scim? ( app-i18n/scim )
+	sdl? (
+		>=media-libs/libsdl2-2.0.0
+		virtual/opengl
```

Comment:
Since you explicitly require `USE=opengl` for `sdl?`, wouldn't it be better to move this into global `opengl?`? Or alternatively, remove the `opengl? ( || ( ... ) )` REQUIRED_USE constraint and ignore the flag when no appropriate backend is enabled.

Reply:
opengl can be used without sdl, keeping `>=media-libs/libsdl2-2.0.0` out does make sense here. I dropped `virtual/opengl` from there though.